bcda/service/resources.go

```diff
@@ -0,0 +1,68 @@
+package service
+
+import "github.com/CMSgov/bcda-app/bcda/constants"
+
+//DataType is used to identify the type of data returned by each resource
+type DataType struct {
+	Adjudicated    bool
+	PreAdjudicated bool
+}
+
+var dataTypeMap map[string]DataType
+
+// SupportsDataType checks if the dataType is supported by the instanced DataType object
+func (r DataType) SupportsDataType(dataType string) bool {
+	switch dataType {
+	case constants.PreAdjudicated:
+		return r.PreAdjudicated
+	case constants.Adjudicated:
+		return r.Adjudicated
+	default:
+		return false
+	}
+}
+
+// init creates a map of ResourceType => DataType configurations to be used by later functions
+func init() {
+	dataTypeMap = map[string]DataType{
+		"Patient":              {Adjudicated: true},
+		"Coverage":             {Adjudicated: true},
+		"ExplanationOfBenefit": {Adjudicated: true},
+		"Observation":          {Adjudicated: true},
+		"Claim":                {PreAdjudicated: true},
+		"ClaimResponse":        {PreAdjudicated: true},
+	}
+}
+
+// GetDataType gets the DataType associated with the given resourceName
+func GetDataType(resourceName string) (DataType, bool) {
+	resource, ok := dataTypeMap[resourceName]
+
+	return resource, ok
+}
+
+// GetDataTypes creates a map of the given resourceNames with their associated DataType objects
+// It returns the resource map and a status flag, with true meaning all resources were found and mapped.
+func GetDataTypes(resourceNames ...string) (map[string]DataType, bool) {
+	foundAll := true
+
+	returnMap := make(map[string]DataType, len(resourceNames))
+
+	if len(resourceNames) == 0 {
+		// If no resource specified, return copy of full map
+		for name, entry := range dataTypeMap {
```

Comment:
Could we replace this block of code with:
returnMap = dataTypeMap

Reply:
Sorry to be late to the conversation, but won't a `returnMap = dataTypeMap` assignment return the original private variable when called?  Meaning, if a downstream method altered the map it would change the value defined in the `init` method?  I'm new to go, but wouldn't it be safer to do the copy?

Reply:
@CosmicSyntax is the above correct or will `returnMap` always be a copy?

Reply:
It is a copy, but a copy of a "smart pointer."  Which means yes, any change down stream would change the globally defined variable. 

This map isn't mutated anywhere, and it's on a hot path. But; on second thought now, I think you're right if you're alluding to future memory safety is worth a little bit of the overhead.

Reply:
In other words: great point Eric and sorry Kyle; let's change this back.